meson.build

```diff
@@ -146,11 +146,86 @@ else
   add_project_arguments('-DDISABLE_ORC=1', language: ['c', 'cpp'])
 endif
 
-# NO Video support
-if get_option('disable-video-support')
-  add_project_arguments('-DNO_VIDEO=1', language: ['c', 'cpp'])
-  message('Disable Video Type Support')
-endif
+
```

Comment:
Isn't this too complicated and lengthy for the given purpose?

Reply:
1. boolean has `true` and `false` but feature has 3 options as below which is making feature handling part more complex although having overlapping flow for each feature. 
   
 - `enabled`: same as required: true
 - `auto`: same as `required: false`, but can be controlled by `auto_features=enabled` to enable all features and disable explicitly only the few they don't want, if any.
 - `disabled`: always return `not_found`

2. With this bit, future features and relative variables can declaratively registered instead of handling flow for every feature. 

3. So, inversion of flow control will making feature pluggable and less error-prone as well as saving LoCs. If we migrate 100% of `enable-*` to registerable feature, it would save 132 - 296 from old meson.build(which expected to be doubled up if we migrate to feature by patching those line due to more complex logic).

Checkout similar strategy used in gst
https://github.com/GStreamer/gst-build/blob/master/meson.build (line 57 -  )

Reply:
I could still go for the old way if requested :)